rockset/resource_scheduled_lambda.go

```diff
@@ -0,0 +1,245 @@
+package rockset
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/rockset/rockset-go-client"
+	"github.com/rockset/rockset-go-client/openapi"
+	"github.com/rockset/rockset-go-client/option"
+)
+
+func resourceScheduledLambda() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a Rockset scheduled lambda, a query lambda that is automatically executed on a schedule.",
+
+		CreateContext: resourceScheduledLambdaCreate,
+		ReadContext:   resourceScheduledLambdaRead,
+		UpdateContext: resourceScheduledLambdaUpdate,
+		DeleteContext: resourceScheduledLambdaDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"rrn": {
+				Description: "RRN of this Scheduled Lambda.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"workspace": {
+				Description:  "Workspace name.",
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: rocksetNameValidator,
+			},
+			"apikey": {
+				Description: "The apikey to use when triggering execution of the associated query lambda.",
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					// apikey is not returned for security reasons
+					return new == ""
+				},
+			},
+			"cron_string": {
+				Description: "The UNIX-formatted cron string for this scheduled query lambda.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"ql_name": {
```

Comment:
this is part of the documentation, so we should be verbose to make it easy to use
```suggestion
			"query_lambda_name": {
```

Reply:
The API uses ql_name, is it better to be verbose or consistent with the API?

Reply:
I think verbose is better, as terraform is self-documenting, and will show the attribute name in outputs - only nerd use the API 😉